### PR TITLE
Plugin generator

### PR DESCRIPTION
**Pull request by alepuccetti**
This PR introduces  Weave Scope plugin generator's code and documentation.

/cc @2opremio

### CODE REVIEW

**Comment by alepuccetti** [comment]
This is relevant to https://github.com/weaveworks/scope/issues/1963

File: README.md [hunk 1]
```diff
@@ -1,2 +1,64 @@
-# scope-plugin-generator
-This is a plugin generator for [Weave Scope](https://github.com/weaveworks/scope)
+# Scope Plugin Generator
+The Scope Plugin Generator uses [Cookiecutter](https://github.com/audreyr/cookiecutter) to automatically generate a [Weave Scope](https://github.com/weaveworks/scope) plugin's basic structure.
```
Comment (2opremio) [inline README.md]: In what language? What can the user expect for the skeleton?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,64 @@
...
+
+You can start developing your own Scope plugins in less then 5 minutes by just following the instructions in the next section.
+
+## Plugin Bootstrapping
+To generate your own plugin you will need Cookiecutter, you can install it following these [instructions](https://cookiecutter.readthedocs.io/en/latest/installation.html).
+
+After you have installed Cookiecutter, you can create a basic Scope plugin structure using the following command: `cookiecutter https://github.com/weaveworks-plugins/scope-plugin-generator`.
+You will be prompted to insert the following basic information about your plugin:
```
Comment (2opremio) [inline README.md]: Are all this all necessary? They seem a bit overwhelming.

If the intention is to get a fast start for users I would only require the absolutely fundamental options and then add an extra section indicating what they may possible when defined.

Also, please see if you can reduce their number.

File: {{cookiecutter.project_root_dir_name}}/circle.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# You need to setup the DOCKER_USER, DOCKER_EMAIL, and DOCKER_PASS
```
Comment (2opremio) [inline {{cookiecutter.project_root_dir_name}}/circle.yml]: Forcing users to use circle seems like a hard requirement.

Comment (alepuccetti) [reply {{cookiecutter.project_root_dir_name}}/circle.yml]: They can use whatever, it seemed nice to provide a boilerplate for the CI/CD too. It is not a requirement, it is more an added bonus.

File: README.md (hunk 1 above)
Comment (alepuccetti) [reply README.md]: What to expect is in the following section. Do you want me to move it up here?

File: README.md (hunk 2 above)
Comment (alepuccetti) [reply README.md]: I can do multiple incremental templates one that generate the code, one for the dockerfile and makefile.

**Comment by alepuccetti** [comment]
@2opremio I address your comments. I removed some attributes that are not essential, and I clarified in the doc the mandatory attributes and the optional ones.
I dropped the multiple templates approach because it was more confusing for the user. Now it is clear that they can just ignore the optional attributes and the `circle.yml` file.

**Pull request closed without merging**